.eslintrc.json

```diff
@@ -1,338 +1,338 @@
+          "Number": "Avoid using the `Number` type. Did you mean `number`?",
+          "Object": "Avoid using the `Object` type. Did you mean `object`?",
+          "String": "Avoid using the `String` type. Did you mean `string`?",
+          "Symbol": "Avoid using the `Symbol` type. Did you mean `symbol`?"
+        }
+      }
+    ],
+    "@typescript-eslint/consistent-type-assertions": [
+      "error",
+      {
+        "assertionStyle": "as",
+        "objectLiteralTypeAssertions": "never"
+      }
+    ],
+    "@typescript-eslint/explicit-function-return-type": "off",
+    "@typescript-eslint/explicit-member-accessibility": [
+      "error",
+      {
+        "accessibility": "explicit",
+        "overrides": {
+          "accessors": "explicit",
+          "constructors": "no-public",
+          "methods": "explicit",
+          "parameterProperties": "off",
+          "properties": "explicit"
+        }
+      }
+    ],
+    "@typescript-eslint/indent": ["off", 4],
+    "@typescript-eslint/member-delimiter-style": [
+      "error",
+      {
+        "multiline": {
+          "delimiter": "semi",
+          "requireLast": true
+        },
+        "singleline": {
+          "delimiter": "semi",
+          "requireLast": false
+        }
+      }
+    ],
+    "@typescript-eslint/member-ordering": [
+      "error",
+      {
+        "default": ["public-instance-method", "public-static-field"]
+      }
+    ],
+    "@typescript-eslint/no-empty-interface": "off",
+    "@typescript-eslint/no-explicit-any": "off",
+    "@typescript-eslint/no-inferrable-types": "off",
+    "@typescript-eslint/no-misused-new": "error",
+    "@typescript-eslint/no-namespace": "error",
+    "@typescript-eslint/no-object-literal-type-assertion": "off",
+    "@typescript-eslint/no-parameter-properties": "error",
+    "@typescript-eslint/no-unused-expressions": "warn",
+    "@typescript-eslint/no-use-before-define": "off",
+    "@typescript-eslint/no-var-requires": "error",
+    "@typescript-eslint/prefer-for-of": "error",
+    "@typescript-eslint/prefer-function-type": "error",
+    "@typescript-eslint/prefer-interface": "off",
+    "@typescript-eslint/triple-slash-reference": "error",
+    "@typescript-eslint/type-annotation-spacing": [
+      "error",
+      {
+        "after": true,
+        "before": false,
+        "overrides": {
+          "arrow": {
+            "after": true,
+            "before": true
+          }
+        }
+      }
+    ],
+    "@typescript-eslint/unified-signatures": "error",
+    "@typescript-eslint/no-unused-vars": "warn",
+    "arrow-parens": "off",
+    "arrow-spacing": "error",
+    "comma-dangle": "off",
+    "complexity": "off",
+    "constructor-super": "error",
+    "curly": "error",
+    "default-case": "error",
+    "dot-notation": "error",
+    "eol-last": "error",
+    "eqeqeq": [
+      "error",
+      "always",
+      {
+        "null": "ignore"
+      }
```

Comment:
I would remove it, it doesn't make sense to allow `==` and `!=` for nulls.

Reply:
It does if you consider this:
https://dorey.github.io/JavaScript-Equality-Table/

null and undefined only coerce to each other and you are unlikely to be explicitly expecting one or the other. Removing this rule will likely introduce a lot of bugs where someone has written `x == null` to catch x being `null` or `undefined`.

Reply:
>Removing this rule will likely introduce a lot of bugs where someone has written x == null to catch x being null or undefined.

Yup, in many cases I would be this `someone` 😃

Reply:
Ok, change reverted, @bartlomiejzuber you can override it after extending this config.

Reply:
> It does if you consider this:
> https://dorey.github.io/JavaScript-Equality-Table/
> 
> null and undefined only coerce to each other and you are unlikely to be explicitly expecting one or the other. Removing this rule will likely introduce a lot of bugs where someone has written `x == null` to catch x being `null` or `undefined`.

The thing is that if you allow `==` for some cases and doesn't allow it for other cases makes code more difficult to read and of course we're not consistent. Over the web & tons articles I've read about JSX/TSX it's very uncommon to use `==`, I would even say it's an official standard to discourage using it mainly because those are very confusing for devs and are very likely to introduce a bug if used in wrong way.

So what I would recommend is to remove this rule and be consistent. If someone is expecting that `x` might be `null` or `undefined` in this case negation should be used e.g. `if (!x)` rather than `if (x == null)`.

If you're afraid that this breaks someone code if they bump the version, you shouldn't because lint will show an error in every place where this 'equality' check was used so it can be replaced easily.

Reply:
Another thing to consider is that disallowing of `==` and `!=` is listed as eslint's recommended config https://eslint.org/docs/rules/ and airbnb eslint config https://github.com/airbnb/javascript#comparison-operators--equality.

That's very uncommon at least for me to see `==` or `!=` being used in the code and in my honest opinion I think that removing this will be beneficent for us. :)

Reply:
I think whilst @hisuwh makes a good point in that this is a safe thing to do from a purist stand point I would wonder if we *should* do this. The reason being that `null` and `undefined` represent two distinct states and given we use strict mode on our types this means we should be explicitly allowing one or both of these values.

I would say it's most likely to be unusual that we should have both in most call signatures / interfaces as we would not be likely to use both states usually it would be along the lines of "I omitted an argument" - `undefined` or "this property has no value" - `null`. So if we were to live by the stricter values it would be preferable to remove this rule.

However, I appreciate all the points made here and written in the format `if (x == null)` vs `if (!x)` are stylistically equivalent (while not functionally) in that there is an implicit understanding that several values resolve ultimately to `true` and it's easy in both cases through lack of awareness to overlook one or many of those scenarios so there's always going to have to be a level of that otherwise we'd have to ban all truthy comparisons e.g. `if (x)`.

@hisuwh @szpght @OwenPattison I think you guys are pro keeping this rule can you give any examples of where something being `null` or `undefined` would be valid?

Reply:
The main reason I raised the point here was due to the significant impact it would have across projects which was not documented as part of the release change notes.

I don't actually mind removing the rule as I think it has the potential to confuse having this one exception for == null, its very easy to have a helper method to === check both things

Reply:
@OwenPattison There's no need to use helper method, you can simply use `if (!x)` which is equivalent of `== null` and I highly encourage to use it. :)

Reply:
> @OwenPattison There's no need to use helper method, you can simply use `if (!x)` which is equivalent of `== null` and I highly encourage to use it. :)

🤣 this is for another conversation on readability!

Reply:
Also subtly different
`!""` === true
`"" == null` === false

Reply:
Ok, since most of us vote "aye" for this rule I pushed it back as a change and updated PR first comment. Please let me know if there is anything else, if not, please leave a review with green "approve" :wink: .